heft-plugins/heft-lint-plugin/src/SarifFormatter.ts

```diff
@@ -0,0 +1,270 @@
+
+      const containsSuppressedMessages: boolean =
+        result.suppressedMessages && result.suppressedMessages.length > 0;
+      const messages: IMessage[] =
+        containsSuppressedMessages && !ignoreSuppressed
+          ? [...result.messages, ...result.suppressedMessages]
+          : result.messages;
+
+      if (messages.length > 0) {
+        for (const message of messages) {
+          const level: string = message.fatal || message.severity === 2 ? 'error' : 'warning';
+          const sarifRepresentation: ISarifRepresentation = {
+            level,
+            message: {
+              text: message.message
+            },
+            locations: [
+              {
+                physicalLocation: {
+                  artifactLocation: {
+                    uri: fileUrl,
+                    index: artifactIndex
+                  }
+                }
+              }
+            ]
+          };
+
+          if (message.ruleId) {
+            sarifRepresentation.ruleId = message.ruleId;
+
+            if (containsSuppressedMessages && !ignoreSuppressed) {
+              sarifRepresentation.suppressions = message.suppressions
+                ? message.suppressions.map((suppression: ISuppressedAnalysis) => {
+                    return {
+                      kind: suppression.kind === 'directive' ? 'inSource' : 'external',
+                      justification: suppression.justification
+                    };
+                  })
+                : [];
+            }
+          } else {
+            sarifRepresentation.descriptor = {
+              id: internalErrorId
+            };
+
+            if (sarifRepresentation.level === 'error') {
+              executionSuccessful = false;
+            }
+          }
+
+          const physicalLocation: ISarifPhysicalLocation = sarifRepresentation.locations[0].physicalLocation;
+
+          if (message.line || message.column) {
+            physicalLocation.region = {};
+            const { line, column, endLine, endColumn } = message;
+            const region: IRegion = {
+              startLine: line ? line : undefined,
+              startColumn: column ? column : undefined,
+              endLine: endLine ? endLine : undefined,
+              endColumn: endColumn ? endColumn : undefined
+            };
+            physicalLocation.region = region;
+          }
+
+          if (message.source) {
+            physicalLocation.region = physicalLocation.region || {};
+            physicalLocation.region.snippet = {
+              text: message.source
+            };
+          }
+
+          if (message.ruleId) {
+            sarifResults.push(sarifRepresentation);
+          } else {
+            toolConfigurationNotifications.push(sarifRepresentation);
+          }
+        }
+      }
+    }
+  }
+
+  const sarifFileValues: string[] = Object.values(sarifFiles);
+  if (sarifFileValues.length > 0) {
+    sarifRun.artifacts = sarifFileValues;
+  }
```

Comment:
```suggestion
  if (sarifFiles.size > 0) {
    sarifRun.artifacts = Array.from(sarifFiles.values());
  }
```

`Object.values()` doesn't work on a `Map`.